packages/ec-linear/src/mutator/umad.rs

```diff
@@ -40,6 +69,15 @@ where
     GeneGenerator: Generator<G::Gene>,
 {
     fn mutate(&self, genome: G, rng: &mut ThreadRng) -> anyhow::Result<G> {
+        if genome.size() == 0 {
+            if let Some(addition_rate) = self.empty_addition_rate {
+                return rng
```

Comment:
This only ever generates one instruction. I actually think we'd probably like this to potentially generate at least a few instructions. A single instruction often won't do anything, and if there's any pressure to move to empty programs (which is probably how we ended up in this situation), a single instruction will probably just get pushed back down to empty pretty quickly.

That said, I'm not sure what a good alternative would be without adding a ton of new parameters. A simple option would be to just have a value that is how many instructions to add. In practice, though, we'd often want that to be a fraction of the maximum number of instructions in a new program, but I don't want to bring that parameter into this. No sure what the best option is.